pkg/ast/spl/spl.peg

```diff
@@ -1977,16 +1977,41 @@ ByClause <- BY fieldList:FieldNameList {
 
 DedupBlock <- PIPE CMD_DEDUP dedupExpr:DedupExpr {
 
+    dedupExp := dedupExpr.(*structs.DedupExpr)
+
     letColReq := &structs.LetColumnsRequest {
-        DedupColRequest: dedupExpr.(*structs.DedupExpr),
+        DedupColRequest: dedupExp,
     }
 
     root := &structs.QueryAggregators {
         PipeCommandType: structs.OutputTransformType,
         OutputTransforms: &structs.OutputTransforms {
             LetColumns: letColReq,
         },
-        DedupExpr: dedupExpr.(*structs.DedupExpr),
+        DedupExpr: dedupExp,
+    }
+
+    if len(dedupExp.DedupSortEles) > 0 {
+        sortExpr := &structs.SortExpr {
+            SortEles: dedupExp.DedupSortEles,
+            SortAscending: dedupExp.DedupSortAscending,
+            Limit: 1000000,
```

Comment:
nit: this is 1 million, but your PR description said 100k. Which do we want?

Reply:
Also, we can handle `dedup sortby` more efficiently than converting it to `sort | dedup`; we may want to do that optimization later. With the way you're doing this, we could end up with too few results (e.g., all the top 1 million rows have the same value, so then only 1 row is returned from dedup, but if the dataset has other rows with different values we should actually return more than 1 row)

Reply:
Yeah I think 1M is fine, yeah we can do the optimization later.